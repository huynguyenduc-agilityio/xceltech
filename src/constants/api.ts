// TODO: Update API after BE update
const END_POINTS = {
  ACCOUNT: '/accounts/',
  LOGIN: '/accounts/login/',
  REGISTER: '/accounts/register/',
  ACTIVE: '/accounts/activate',
  USER_DETAIL: '/accounts/me/',
  LEAVES: '/leave-applications/',
  USER_LEAVES: '/leave-applications/users',
  DOCUMENT_PROFILE: '/documents/',
  NOTIFICATIONS: '/notifications/',
  JOB: '/jobs/',
};

export { END_POINTS };
