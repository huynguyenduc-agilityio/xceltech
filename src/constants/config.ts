export const REACT_QUERY = {
  STALE_TIME: 30000,
};
