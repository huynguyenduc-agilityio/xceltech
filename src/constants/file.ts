export enum FileType {
  Pdf = 'pdf',
  Csv = 'csv',
  Excel = 'excel',
}
