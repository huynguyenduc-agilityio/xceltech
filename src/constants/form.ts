export const LEAVE_TYPE_OPTIONS = [
  { value: 'annual leave', label: 'Annual Leave' },
  { value: 'sick', label: 'Sick' },
  { value: 'exam', label: 'Exam' },
  { value: 'maternity', label: 'Maternity' },
];

export const RELIEF_OFFICER_OPTIONS = [
  { value: 'June Heaney', label: 'June Heaney' },
  { value: 'Rudy Witting', label: 'Rudy Witting' },
  { value: 'Elaine Prohaska', label: 'Elaine Prohaska' },
  { value: 'Jordan Corwin', label: 'Jordan Corwin' },
  { value: 'Nadine Collins', label: 'Nadine Collins' },
  { value: 'Caleb Thiel', label: 'Caleb Thiel' },
];

//TODO: add more department
export const DEPARTMENT_OPTIONS = [
  { value: 'designMarketing', label: 'Design & Marketing' },
];

//TODO: add more job
export const JOB_TITLE_OPTIONS = [
  { value: 'designer', label: 'UI / UX Designer' },
];

export const JOB_CATEGORY_OPTIONS = [
  { value: 'fulltime', label: 'Full time' },
  { value: 'parttime', label: 'Part time' },
];
