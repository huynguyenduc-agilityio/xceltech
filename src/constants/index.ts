export * from './navigation';
export * from './unit';
export * from './regex';
export * from './pagination';
export * from './toast';
export * from './form';
export * from './router';
export * from './api';
export * from './config';
export * from './localStorageKey';
export * from './queries';
export * from './field';
export * from './file';
export * from './images';
export { default as MESSAGES } from './message';
