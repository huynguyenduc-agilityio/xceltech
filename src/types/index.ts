export * from './status';
export * from './employee';
export * from './table';
export * from './auth';
export * from './form';
export * from './tab';
export * from './user';
export * from './queryKey';
export * from './leave';
export * from './filter';
export * from './error';
export * from './notification';
export * from './profile';
