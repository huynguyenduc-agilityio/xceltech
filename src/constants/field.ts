import { IEmployeeGuarantor } from '@/types';

type FormFieldKey<T> = Exclude<keyof T, 'id'>;

type BaseField<T> = {
  name: FormFieldKey<T>;
  label: string;
  placeholder: string;
};

export const BASE_GUARANTOR_FIELD: BaseField<IEmployeeGuarantor>[] = [
  {
    name: 'name',
    label: 'Guarantor’s Name',
    placeholder: 'Enter your guarantor’s name',
  },
  {
    name: 'job',
    label: 'Job Title / Occupation',
    placeholder: 'Enter your job title / occupation',
  },
  {
    name: 'phone',
    label: 'Phone No',
    placeholder: 'Enter your phone number',
  },
];
