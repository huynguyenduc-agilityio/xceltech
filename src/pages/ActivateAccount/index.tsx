import { useNavigate, useSearchParams } from 'react-router-dom';
import { CircleCheck } from 'lucide-react';

// Constants
import { AUTHENTICATION_PAGE, MESSAGES } from '@/constants';

// Types
import { ToastStatus } from '@/types';

// Hooks
import { useActivateAccount, useToast } from '@/hooks';

// Components
import { Button } from '@/components';

const ActivateAccount = () => {
  const navigate = useNavigate();
  const { toast } = useToast();
  const [searchParams] = useSearchParams();
  const uidb64 = searchParams.get('uidb64');
  const token = searchParams.get('token');

  const { handleActivateUser, isActivateLoading } = useActivateAccount({
    uidb64,
    token,
  });

  const handleActivate = async () => {
    if (!uidb64 || !token) {
      toast({ status: ToastStatus.Error, title: 'Invalid activation' });
      return;
    }

    try {
      await handleActivateUser();

      toast({
        status: ToastStatus.Success,
        title: MESSAGES.AUTHENTICATION.ACTIVATE_SUCCESS,
      });

      navigate(AUTHENTICATION_PAGE.USER_SIGN_IN);
    } catch {
      toast({
        status: ToastStatus.Error,
        title: MESSAGES.AUTHENTICATION.ACTIVATE_FAILED,
      });
    }
  };

  return (
    <div className="flex items-center justify-center h-screen bg-gray-100">
      <div className="text-center bg-white p-8 rounded-lg shadow-lg">
        <div className="flex items-center gap-4">
          <CircleCheck className="mx-auto text-black-default w-12 h-12" />
          <h1 className="text-2xl font-bold">Activate your account</h1>
        </div>
        <p className="text-lg mt-8">Click the button below</p>
        <Button
          className="w-[150px] h-[54px] text-lg mt-4 py-3 "
          onClick={handleActivate}
          isLoading={isActivateLoading}
        >
          ACTIVATE
        </Button>
      </div>
    </div>
  );
};

export default ActivateAccount;
