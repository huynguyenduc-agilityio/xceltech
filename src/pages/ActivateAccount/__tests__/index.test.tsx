import { render, screen, fireEvent, waitFor } from '@testing-library/react';

// Constants
import { AUTHENTICATION_PAGE, MESSAGES } from '@/constants';

// Types
import { ToastStatus } from '@/types';

import ActivateAccount from '..';

let mockSearchParams = new URLSearchParams({
  uidb64: 'mockUid',
  token: 'mockToken',
});
const mockNavigate = jest.fn();
const mockToast = jest.fn();
const mockHandleActivateUser = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
  useSearchParams: () => [mockSearchParams],
}));

jest.mock('@/hooks', () => ({
  useToast: () => ({ toast: mockToast }),
  useActivateAccount: () => ({
    handleActivateUser: mockHandleActivateUser,
    isActivateLoading: false,
  }),
}));

describe('ActivateAccount', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockSearchParams = new URLSearchParams({
      uidb64: 'mockUid',
      token: 'mockToken',
    });
  });

  it('renders correctly and matches snapshot', () => {
    const { container } = render(<ActivateAccount />);
    expect(container).toMatchSnapshot();
  });

  it('activates account successfully and navigates', async () => {
    mockHandleActivateUser.mockResolvedValueOnce(undefined);

    render(<ActivateAccount />);
    fireEvent.click(screen.getByRole('button', { name: /activate/i }));

    await waitFor(() => {
      expect(mockHandleActivateUser).toHaveBeenCalled();
      expect(mockToast).toHaveBeenCalledWith({
        status: ToastStatus.Success,
        title: MESSAGES.AUTHENTICATION.ACTIVATE_SUCCESS,
      });
      expect(mockNavigate).toHaveBeenCalledWith(
        AUTHENTICATION_PAGE.USER_SIGN_IN,
      );
    });
  });

  it('shows error toast on activation failure', async () => {
    mockHandleActivateUser.mockRejectedValueOnce(new Error('Failed'));

    render(<ActivateAccount />);
    fireEvent.click(screen.getByRole('button', { name: /activate/i }));

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalledWith({
        status: ToastStatus.Error,
        title: MESSAGES.AUTHENTICATION.ACTIVATE_FAILED,
      });
      expect(mockNavigate).not.toHaveBeenCalled();
    });
  });

  it('shows error toast when uidb64 or token is missing', async () => {
    mockSearchParams = new URLSearchParams();

    render(<ActivateAccount />);
    fireEvent.click(screen.getByRole('button', { name: /activate/i }));

    await waitFor(() => {
      expect(mockToast).toHaveBeenCalledWith({
        status: ToastStatus.Error,
        title: 'Invalid activation',
      });
      expect(mockHandleActivateUser).not.toHaveBeenCalled();
    });
  });
});
