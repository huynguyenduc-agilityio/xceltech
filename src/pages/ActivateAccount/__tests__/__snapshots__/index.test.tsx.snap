// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivateAccount renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-center h-screen bg-gray-100"
  >
    <div
      class="text-center bg-white p-8 rounded-lg shadow-lg"
    >
      <div
        class="flex items-center gap-4"
      >
        <svg
          class="lucide lucide-circle-check mx-auto text-black-default w-12 h-12"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="m9 12 2 2 4-4"
          />
        </svg>
        <h1
          class="text-2xl font-bold"
        >
          Activate your account
        </h1>
      </div>
      <p
        class="text-lg mt-8"
      >
        Click the button below
      </p>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white disabled:hover:bg-primary w-[150px] h-[54px] text-lg mt-4 py-3"
      >
        ACTIVATE
      </button>
    </div>
  </div>
</div>
`;
